### PR TITLE
NIFI-606 Add a NiFi Storm Spout 

### PR DESCRIPTION
**Pull request by bbende**
Adding a NiFiSpout for integration with Apache Storm, very similar to the NiFiReceiver for Apache Spark.

In order to use this:
* Run mvn clean install on this branch so the spout is installed in your Maven repository
* Download Storm 0.9.4 and add  Storm to your path
* Create a project that brings in the spout as a dependency and builds an uber jar, an example project is here: https://github.com/bbende/nifi-storm-topology
* Start NiFi and create a flow that brings data to an Output Port named "Data for Storm"
* Run the topology: storm jar yourproject.jar org.example.YourTopology

### CODE REVIEW

File: nifi/nifi-external/nifi-storm-spout/src/main/java/org/apache/nifi/storm/NiFiSpout.java [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+                        final List<NiFiDataPacket> dataPackets = new ArrayList<>();
+                        do {
+                            // Read the data into a byte array and wrap it along with the attributes
+                            // into a NiFiDataPacket.
+                            final InputStream inStream = dataPacket.getData();
+                            final byte[] data = new byte[(int) dataPacket.getSize()];
+                            StreamUtils.fillBuffer(inStream, data);
+
+                            final Map<String, String> attributes = dataPacket.getAttributes();
+                            final NiFiDataPacket NiFiDataPacket = new NiFiDataPacket() {
```
Comment (apiri) [inline nifi/nifi-external/nifi-storm-spout/src/main/java/org/apache/nifi/storm/NiFiSpout.java]: Minor:  For the sake of consistency, your variable should follow camel casing.

**Comment by apiri** [comment]
Bryan, good stuff.

Thanks for the detailed instructions and samples to get this tested in an appreciable fashion with Storm. 

Minor points:  Camel casing of the NiFiDataPacket variable.
Also, since this PR was submitted, and due to the number of contributions and features under way, the develop branch has since been migrated to 0.2.0.  

If you would like, I do not mind I can make these changes for your patch upon merge, but if you would prefer to do so yourself and update the PR, that also works.

Let me know your thoughts.

+1

**Comment by bbende** [comment]
Aldrin,

Thanks for reviewing. I'm good with you making the updates when you merge
it, appreciate it!

-Bryan

On Friday, June 5, 2015, Aldrin Piri <notifications@github.com> wrote:

> Bryan, good stuff.
>
> Thanks for the detailed instructions and samples to get this tested in an
> appreciable fashion with Storm.
>
> Minor points: Camel casing of the NiFiDataPacket variable.
> Also, since this PR was submitted, and due to the number of contributions
> and features under way, the develop branch has since been migrated to
> 0.2.0.
>
> If you would like, I do not mind I can make these changes for your patch
> upon merge, but if you would prefer to do so yourself and update the PR,
> that also works.
>
> Let me know your thoughts.
>
> +1
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/apache/incubator-nifi/pull/61#issuecomment-109358436>.
>


-- 
Sent from Gmail Mobile

**Comment by apiri** [comment]
960560723dc4ed8968c1a4d072a5b4902c371298 resolves #61 and NIFI-606

**Comment by apiri** [comment]
@bbende 
Forgot to include the magic words in the commit and couldn't trick Github, so if you could close this out when you have the opportunity, it would be appreciated.  The backing JIRA issue has been resolved.

**Pull request closed without merging**